contrib/mod_sftp/auth-hostbased.c

```diff
@@ -94,6 +94,16 @@ int sftp_auth_hostbased(struct ssh2_packet *pkt, cmd_rec *pass_cmd,
   if (strncmp(hostkey_algo, "ssh-rsa", 8) == 0) {
     pubkey_type = SFTP_KEY_RSA;
 
+#ifdef HAVE_SHA256_OPENSSL
+  } else if (strncmp(hostkey_algo, "rsa-sha2-256", 13) == 0) {
+    pubkey_type = SFTP_KEY_RSA_SHA256;
+#endif /* HAVE_SHA256_OPENSSL */
+
+#ifdef HAVE_SHA512_OPENSSL
+  } else if (strncmp(hostkey_algo, "rsa-sha2-512", 13) == 0) {
+    pubkey_type = SFTP_KEY_RSA_SHA512;
+#endif /* HAVE_SHA512_OPENSSL */
```

Comment:
This was missed on the initial implementation.  Not many sites use hostkey authentication, but better to be thorough.